src/main/io/gps.c

```diff
@@ -528,6 +522,40 @@ void gpsUpdate(timeUs_t currentTimeUs)
                 // remove GPS from capability
                 sensorsClear(SENSOR_GPS);
                 gpsSetState(GPS_LOST_COMMUNICATION);
+            } else {
+                switch (gpsData.messageState) {
+                    case GPS_MESSAGE_STATE_IDLE:
+                        if (!gpsData.airborneMode && ARMING_FLAG(ARMED)) {
+                            gpsData.messageState = GPS_MESSAGE_STATE_SETTING_AIRBORNE;
+                            gpsData.state_position = 0;
+                        }
+                        if (gpsData.airborneMode && !ARMING_FLAG(ARMED)) {
+                            gpsData.messageState = GPS_MESSAGE_STATE_SETTING_PEDESTRIAN;
+                            gpsData.state_position = 0;
+                        }
+                        break;
+                    case GPS_MESSAGE_STATE_SETTING_AIRBORNE:
+                        if (gpsData.state_position < sizeof(ubloxAirborne)) {
+                            serialWrite(gpsPort, ubloxAirborne[gpsData.state_position]);
+                            gpsData.state_position++;
+                        } else {
+                            gpsData.airborneMode = true;
+                            gpsData.messageState = GPS_MESSAGE_STATE_IDLE;
+                        }
+                        break;
+                    case GPS_MESSAGE_STATE_SETTING_PEDESTRIAN:
+                        //We only want to send the first message of the Init chunk, which takes exactly as many bytes as setting Airborne
+                        if (gpsData.state_position < sizeof(ubloxAirborne)) {
+                            serialWrite(gpsPort, ubloxInit[gpsData.state_position]);
+                            gpsData.state_position++;
+                        } else {
+                            gpsData.airborneMode = true;
```

Comment:
This will cause loop until target is armed

Reply:
Oops, good catch!